cdap-kms/src/main/java/co/cask/cdap/security/store/KMSSecureStore.java

```diff
@@ -0,0 +1,148 @@
+  }
+
+  /**
+   * Stores an element in the secure store. The key is stored as namespace:name in the backing store,
+   * assuming ":" to be the separator.
+   * @param namespace The namespace this key belongs to.
+   * @param name Name of the element to store.
+   * @param data The data that needs to be securely stored.
+   * @param description User provided description of the entry.
+   * @param properties Metadata associated with the data
+   * @throws IOException If it failed to store the key in the store.
+   */
+  @Override
+  public void put(String namespace, String name, byte[] data, String description, Map<String, String> properties)
+    throws IOException {
+    KeyProvider.Options options = new KeyProvider.Options(conf);
+    options.setDescription(description);
+    options.setAttributes(properties);
+    options.setBitLength(data.length * Byte.SIZE);
+    try {
+      provider.createKey(SecureStoreUtils.getKeyName(namespace, name), data, options);
+    } catch (IOException e) {
+      throw new IOException("Failed to store the key. ", e);
+    }
+  }
+
+  /**
+   * Deletes the element with the given name.
+   * @param namespace The namespace this key belongs to.
+   * @param name Name of the element to be deleted.
+   */
+  @Override
+  public void delete(String namespace, String name) throws IOException {
+    try {
+      provider.deleteKey(SecureStoreUtils.getKeyName(namespace, name));
+    } catch (IOException e) {
+      throw new IOException("Failed to delete the key. ", e);
+    }
+  }
+
+  /**
+   * List of all the entries in the secure store.
+   * @return A list of {@link SecureStoreMetadata} objects representing the data stored in the store.
+   * @param namespace The namespace this key belongs to.
+   */
+  @Override
+  public List<SecureStoreMetadata> list(String namespace) throws IOException {
+    String prefix = namespace + SecureStoreUtils.NAME_SEPARATOR;
+    List<String> keysInNamespace = new ArrayList<>();
+    KeyProvider.Metadata[] metadatas;
+    List<SecureStoreMetadata> secureStoreMetadatas;
+    try {
+      for (String key : provider.getKeys()) {
+        if (key.startsWith(prefix)) {
+          keysInNamespace.add(key);
+        }
+      }
+      metadatas = provider.getKeysMetadata(keysInNamespace.toArray(new String[0]));
+      secureStoreMetadatas = new ArrayList<>();
+    } catch (IOException e) {
+      throw new IOException("Failed to get the list of elements from the secure store.", e);
+    }
+    for (int i = 0; i < metadatas.length; i++) {
+      KeyProvider.Metadata metadata = metadatas[i];
+      SecureStoreMetadata meta = SecureStoreMetadata.of(keysInNamespace.get(i).substring(prefix.length()),
+                                                        metadata.getDescription(),
+                                                        metadata.getAttributes());
+      secureStoreMetadatas.add(meta);
+    }
+    return secureStoreMetadatas;
```

Comment:
This will be what's returned by the REST handler for example, right? Good to have a guard?